### PR TITLE
dxvk: fix DLL override setup

### PR DESCRIPTION
**Pull request by 9ary**
###### Description of changes

`basename srcpath` simply prints "srcpath" which results in a single DLL override for srcpath.dll to be installed... clearly not what we want.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ckiee** [comment]
#### Manual test for x86_64-linux
```
~/git/nixpkgs -> DXVK_HUD=1 NIX_PATH=nixpkgs=. WINEPREFIX=$(realpath meow) nix-shell -p wineWowPackages.stable -p dxvk
[nix-shell:~/git/nixpkgs]$ winecfg *close GUI with no changes*
[nix-shell:~/git/nixpkgs]$ wine /path/to/Unigine_Heaven-4.0.exe *install*
[nix-shell:~/git/nixpkgs]$ wine explorer *launch .bat in program files x86, start benchmark, verify no HUD*
[nix-shell:~/git/nixpkgs]$ setup_dxvk.sh install
/nix/store/k9nwgwa6yvkpcm51xl8mf1iy9h8j39ic-dxvk/bin/setup_dxvk.sh: line 45: $1: unbound variable
[nix-shell:~/git/nixpkgs]$ wine explorer *launch .bat in program files x86, start benchmark, verify HUD is there*
```
That last step ("verify HUD is there") is what fails in the commit this branch is based on, and this PR indeed fixes it, <br>
Screenshot of Unigine Heaven running under wine with the DXVK HUD visible

#### Tangential to this PR
I *really* wish we had a NixOS test for this, though the infra to build a d3d11-triangle.exe-esque thing might be missing.

For anyone willing to take that on, [d3d11-triangle](https://github.com/doitsujin/dxvk/blob/0e0ee61d9b879e3ea066326874b0f7aad6f6f18e/tests/d3d11/test_d3d11_triangle.cpp) is a little demo that used to ship with dxvk (that commit is from 2018!) and I suspect it would be an easy target for the NixOS test runner's OCR.

File: pkgs/misc/dxvk/setup_dxvk.sh [hunk 1]
```diff
@@ -261,5 +261,5 @@ done
```
Comment (ckiee) [inline pkgs/misc/dxvk/setup_dxvk.sh:261]: Running the script with no arguments yields:
```
[nix-shell:~/git/nixpkgs]$ setup_dxvk.sh
/nix/store/yxi50hdrs81bkxz1wqgmgyidw57rhnrl-dxvk/bin/setup_dxvk.sh: line 45: $1: unbound variable
```
Could you fix that while we're here?

Comment (9ary) [reply pkgs/misc/dxvk/setup_dxvk.sh:261]: Sure that shouldn't be too hard.

Comment (9ary) [reply pkgs/misc/dxvk/setup_dxvk.sh:262]: Done, went through with shellcheck and fixed all warnings while I was at it.

**Comment by reckenrode** [comment]
Thanks for the patch and review. The latest commits (including the one fixing the shellcheck warnings) LGTM.

Regarding possible NixOS tests, I assume the triangle demo could be built and exposed as a separate output, which the test could access via `pkgsCross`.

**Review by reckenrode — APPROVED** [review]

File: pkgs/misc/dxvk/setup_dxvk.sh (hunk 1 above)
Comment (ckiee) [reply pkgs/misc/dxvk/setup_dxvk.sh:262]: Your fix for this is so elegant! I wouldn't have thought of that, really nice!

([commit](https://github.com/NixOS/nixpkgs/pull/220824/commits/f67a9f0e27948fd2f16e5027c401fee80b1fef64))

Comment (9ary) [reply pkgs/misc/dxvk/setup_dxvk.sh:262]: I can't really take credit for that unfortunately, the same construct was used elsewhere in the script. :P

Comment (ckiee) [reply pkgs/misc/dxvk/setup_dxvk.sh:262]: Oh well, learned something new :purple_heart:

**Review by ckiee — APPROVED** [review]
Manual test passes again, code looks good though I haven't manually looked at all the shell semantics, but shellcheck should be good.

**Pull request merged**